R/app_alphaDiv.R

```diff
@@ -5,22 +5,29 @@
 #' @param df data.table with rows corresponding to records, one column indicating the record id, and all other columns corresponding to taxa abundance.
 #' @param recordIdColumn string defining the name of the df column that specifies record ids. Note, all other columns must be numeric and will be treated as abundance values.
 #' @param method string defining the the alpha diversity method. Accepted values are 'shannon','simpson', and 'evenness'
+#' @param naToZero boolean indicating if NAs in numeric columns should be replaced with 0
 #' @param verbose boolean indicating if timed logging is desired
 #' @return data.table with columns recordIdColumn, "alphaDiversity".
 #' @importFrom vegan diversity
 #' @importFrom stringi stri_trans_totitle
 #' @import veupathUtils
 #' @import data.table
 #' @export
-alphaDiv <- function(df, recordIdColumn, method = c('shannon','simpson','evenness'), verbose = c(TRUE, FALSE)) {
+alphaDiv <- function(df, recordIdColumn, method = c('shannon','simpson','evenness'), naToZero = c(FALSE, TRUE), verbose = c(TRUE, FALSE)) {
 
     # Initialize and check inputs
     method <- veupathUtils::matchArg(method)
+    naToZero <- veupathUtils::matchArg(naToZero)
     verbose <- veupathUtils::matchArg(verbose)
 
     computeMessage <- ''
     veupathUtils::logWithTime(paste("Received df table with", nrow(df), "samples and", (ncol(df)-1), "taxa."), verbose)
     
+    if (naToZero) {
+      # Replace NA values with 0
+      veupathUtils::setNaToZero(df, cols = colnames(df[, -..recordIdColumn]))
```

Comment:
seems like you could just not pass cols as an arg here, given the default behavior of setNaToZero? that might even be clearer as i imagine the reader would just assume it was applied across the whole df..